### ISSUE TITLE
Az.Synapse Set-AzSynapsePipeline - Ignores literal nulls (value = null) in json definition

### ISSUE THREAD

**Issue by mwc360** [Azure/azure-powershell#16397]
### Description

This module has a bug when using the Set- cmdlets to publish json definition objects to Synapse.

If the JSON source file contains elements specified as literal nulls when the definition is published to Synapse it ignores the JSON elements that were literal nulls (i.e. value = null). This causes the use of the 'Treat as Null' Stored Procedure activity to not be publishable from the json files. I was able to work around this via directly using the Synapse Rest APIs, the same files published via the APIs do not strip away any literal null json element.

### Issue script & Debug output

```PowerShell
n/a
```

### Environment data

```PowerShell
5.1
```

### Module versions

```PowerShell
I tested via versions 0.1.0, 0.7.0, 0.18.0, the bug occurs with all versions
```

### Error output

```PowerShell
No error output from Powershell, the bug is that the published result to Synapse will ignore any json element set to equal a literal null (i.e. value = null)
```

**Comment by dingmeng-xue** [comment]
Thank you for your reporting. Tagging and routing to the team member best able to assist.

**Comment by wonner** [comment]
Hi @mwc360 , Thanks for your feedback. I can reproduce this problem. I would like to know will it cause any problems if it strips away the literal null?

**Comment by mwc360** [comment]
@wonner Yes, it undoes the 'Treat as Null' feature (checkbox on the parameter mapping pane) for the Stored Procedure and other activities where you map values or expressions to parameters. If the 'Treat as Null' checkbox was checked as true in ADF (value = null in JSON) it results in the box being unchecked and no value being passed into the parameter, however since the parameter is still defined it will result in the activity failing as you cannot reference a parameter but not map any value into the parameter.

**Comment by wonner** [comment]
Hi @mwc360 , We are looking into it and will get back to you as soon as we have an update. Sorry for your inconvenience.

**Comment by arcticdba** [comment]
Can confirm this is STILL an issue in version 3 (the one currently in the cloud shell in Azure).

**Comment by mikegoatly** [comment]
I'm pretty sure the underlying issue is in the `Azure.Analytics.Synapse.Artifacts` library that's being used. The `Write` definition for `ScriptActivityParameter` handles the Value property like this: ([source](https://github.com/Azure/azure-sdk-for-net/blob/046dcb58385908041e4c7b4778bcb7a0866fc4f4/sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/Generated/Models/ScriptActivityParameter.Serialization.cs#L31))

``` csharp
if (Optional.IsDefined(Value))
{
    writer.WritePropertyName("value"u8);
    JsonSerializer.Serialize(writer, Value);
}
```

Where `Optional.IsDefined` will return `false` if the value is `null`, meaning that the property won't be emitted.

I couldn't find an existing issue for it, so I've raised an issue for the Azure SDK team here: https://github.com/Azure/azure-sdk-for-net/issues/40149
